packages/webview_flutter/webview_flutter_android/lib/src/android_webview_controller.dart

```diff
@@ -766,6 +766,15 @@ class AndroidNavigationDelegate extends PlatformNavigationDelegate {
           weakThis.target!._handleNavigation(url, isForMainFrame: true);
         }
       },
+      doUpdateVisitedHistory: (
+        android_webview.WebView webView,
+        String url,
+        bool isReload,
+      ) {
+        if (weakThis.target?._onUrlChange != null) {
+          weakThis.target!._onUrlChange!(UrlChange(url: url));
```

Comment:
You don't want to implement the subclassing example you showed in your comment, and adding `isReload` here?